nat-lab/tests/test_mesh_firewall.py

```diff
@@ -359,8 +531,21 @@ async def test_mesh_firewall_file_share_port(
             allow_peer_send_files=allow_peer_send_file,
         )
 
-        CLIENT_ALPHA_IP = alpha.ip_addresses[0]
-        CLIENT_BETA_IP = beta.ip_addresses[0]
+        CLIENT_PROTO = IPProto.IPv4
+        CLIENT_ALPHA_IP = ""
+        CLIENT_BETA_IP = ""
+
+        if alpha_ip_stack in [IPStack.IPv4, IPStack.IPv4v6]:
```

Comment:
This should not be required :thinking:  You should be able to do:
```python
CLIENT_ALPHA_IP = testing.unpack_optional(alpha.get_ip_address(alpha_ip_stack))
```

Reply:
`IPStack` and `IPProto` are not the same. `get_ip_address(IPProto)`.

Reply:
In any case, we should somehow avoid duplication caused by stack enumeration (IPv4/IPv6), having that method is just one of the ways to achieve less duplication :)

Reply:
Than we should decide, which test/address/etc to use, when there's IPv4v6 configuration.

Reply:
Somewhat fixed it :smile: